src/fluid/braginskii/bragViscosity.cpp

```diff
@@ -0,0 +1,110 @@
+// ***********************************************************************************
+// Idefix MHD astrophysical code
+// Copyright(C) Geoffroy R. J. Lesur <geoffroy.lesur@univ-grenoble-alpes.fr>
+// and other code contributors
+// Licensed under CeCILL 2.1 License, see COPYING for more information
+// ***********************************************************************************
+
+// This source code is largely inspired from the viscous_flux of Pluto4.2
+// ((c) P. Tzeferacos & A. Mignone)
+
+// Implementation of monotonicity-preserving viscous flux following ZuHone et al.,
+// ApJ
+
+#include <string>
+
+#include "idefix.hpp"
+#include "bragViscosity.hpp"
+#include "dataBlock.hpp"
+#include "fluid.hpp"
+
+
+
+
+////or what should be the exact equivalent, but for some reason isn't, maybe ask Geoffroy why:
```

Comment:
Averaging 2 consecutive cell centered B is not equivalent to taking a face-centered B.

The rule is to use the same number of averages in every direction, to avoid introducing an anisotropic numerical diffusion. So I construct everything on cell-centered field, even when it means reconstructing a face-averaged field.

Reply:
Ok I see, this was a question I had for you since a long time so thanks for the enlightenment.